backend/src/main/java/com/github/moinmarcell/backend/security/SecurityConfig.java

```diff
@@ -0,0 +1,32 @@
+package com.github.moinmarcell.backend.security;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.crypto.argon2.Argon2PasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+
+@EnableWebSecurity
+public class SecurityConfig {
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new Argon2PasswordEncoder();
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity) throws Exception {
+        return httpSecurity
+                .csrf()
+                .csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
+                .and()
+                .httpBasic().and()
+                .authorizeRequests()
+                .antMatchers("/my-profile").authenticated()
+                .antMatchers("/**").permitAll()
```

Comment:
For later: 
 Try working with an allow-list, not a deny list.
 If you block everything except the endpoints you opened, you can't forget to block access to new secured endpoints.
```suggestion
                .antMatchers("/").permitAll()
                .antMatchers("/**").authenticated()
```